extensions/parquet/table/src/main/java/io/deephaven/parquet/table/ParquetTools.java

```diff
@@ -890,20 +1002,58 @@ private static Map<String, Map<ParquetCacheTags, Object>> buildComputedCache(
      *        will be computed on demand.
+     * @deprecated Use {@link #writeTables(Table[], String[], ParquetInstructions, String[][])} instead with
+     *             {@link TableDefinition} provided through {@link ParquetInstructions.Builder#setTableDefinition}.
      */
+    @Deprecated
     public static void writeParquetTables(
             @NotNull final Table[] sources,
             @NotNull final TableDefinition definition,
             @NotNull final ParquetInstructions writeInstructions,
             @NotNull final File[] destinations,
             @Nullable final String[][] indexColumnArr) {
+        final String[] destinationPaths = Arrays.stream(destinations).map(File::getPath).toArray(String[]::new);
+        writeTables(sources, destinationPaths, ensureTableDefinition(writeInstructions, definition), indexColumnArr);
+    }
+
+    /**
+     * Writes tables to disk in parquet format to a supplied set of destinations. The {@link TableDefinition} to use for
+     * writing must be provided as part of {@link ParquetInstructions}.
+     *
+     * @param sources The tables to write
+     * @param destinations The destination paths or URIs. Any non-existing directories in the paths provided are
+     *        created. If there is an error, any intermediate directories previously created are removed; note this
+     *        makes this method unsafe for concurrent use.
+     * @param writeInstructions Write instructions for customizations while writing
+     * @param indexColumnArr Arrays containing the column names for indexes to persist. The write operation will store
+     *        the index info as sidecar tables. This argument is used to narrow the set of indexes to write, or to be
+     *        explicit about the expected set of indexes present on all sources. Indexes that are specified but missing
+     *        will be computed on demand.
+     */
+    public static void writeTables(
+            @NotNull final Table[] sources,
+            @NotNull final String[] destinations,
+            @NotNull final ParquetInstructions writeInstructions,
+            @Nullable final String[][] indexColumnArr) {
+        final TableDefinition definition = writeInstructions.getTableDefinition().orElseThrow(
+                () -> new IllegalArgumentException("Table definition must be provided"));
+        final File[] destinationFiles = new File[destinations.length];
+        for (int i = 0; i < destinations.length; i++) {
+            final URI destinationURI = convertToURI(destinations[i], false);
+            if (!FILE_URI_SCHEME.equals(destinationURI.getScheme())) {
+                throw new IllegalArgumentException(
+                        "Only file URI scheme is supported for writing parquet files, found" +
+                                "non-file URI: " + destinations[i]);
+            }
+            destinationFiles[i] = new File(destinationURI);
+        }
```

Comment:
While unfortunate, this is probably the right approach to future-proof our API for object store support.